core_api/src/app.py

```diff
@@ -112,38 +111,22 @@ def health():
 
 
 @app.post("/file", response_model=File, tags=["file"])
-async def create_upload_file(file: UploadFile, ingest: bool = True) -> File:
+async def create_upload_file(name: str, type: str, location: str, ingest=True) -> File:
     """Upload a file to the object store and create a record in the database
 
     Args:
-        file (UploadFile): The file to upload
+        name (str): The file name to be recorded
+        type (str): The file type to be recorded
+        location (str): The presigned file resource location
```

Comment:
Is there a better data type for the location than 'str'? I couldn't import anything useful from Pydantic

Reply:
https://docs.pydantic.dev/2.0/usage/types/file_types/ ?